orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/persistence/ExecutionRepository.java

```diff
@@ -156,6 +156,15 @@ Execution retrievePipelineForCorrelationId(@Nonnull String correlationId)
 
   List<String> retrieveAllExecutionIds(@Nonnull ExecutionType type);
 
+  @Nullable
+  default String getPartition() {
+    throw new UnsupportedOperationException();
```

Comment:
should the default impl return `null`? that way default `handlesPartition` doesn't blow up?

Reply:
you're probably right, I was wondering about null vs unsupported